src/selection.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  InlineAdjacentChanges,
+  trackChangesPluginKey,
+} from '@manuscripts/track-changes-plugin'
+import { ManuscriptEditorState } from '@manuscripts/transform'
+import { Node, ResolvedPos } from 'prosemirror-model'
+import { Selection } from 'prosemirror-state'
+import { Mappable } from 'prosemirror-transform'
+
+import { isTextSelection } from './commands'
+
+/**
+ * Selecting multiple nodes without moving selection cursor
+ */
+export class NodesSelection extends Selection {
```

Comment:
Is this class used somewhere? I only see the definition but no usages.

Reply:
will use it in article-editor to select multiple nodes, as we can't select multiple nodes with [pm-node-selection](https://github.com/ProseMirror/prosemirror-state/blob/09ffa0c8b5a331f6fbec2809b5bd454e4a43744d/src/selection.ts#L325) also text-selection will not work for that case as cursor will be moved